lib/chef/search/query.rb

```diff
@@ -89,7 +89,7 @@ def search(type, query='*:*', *args, &block)
         if block
           response["rows"].each { |row| block.call(row) if row }
           unless (response["start"] + response["rows"].length) >= response["total"]
```

Comment:
Since this PR was already merged, you're unlikely to have many people see this comment, if any more than me.

If you have a test case that you could reproduce where this would be a problem, I'd recommend starting with creating an issue. If you're just thinking, I'd go to the [chef](http://lists.opscode.com/sympa/info/chef) or [chef-dev](http://lists.opscode.com/sympa/info/chef-dev) mailing list.